RabbitMQ.Stream.Client/Connection.cs

```diff
@@ -118,6 +118,7 @@ private static System.IO.Stream MaybeTcpUpgrade(NetworkStream networkStream, Ssl
 
         private async Task ProcessIncomingFrames()
         {
+            Exception caught = null;
```

Comment:
Why do we need this?

Reply:
It's so that we don't lose any exceptions that may happen in the frame reader task. `caught` is passed to `CompleteAsync`. I've got a couple other things to add, so I moved this back to a draft.